internal-enrichment/safebrowsing/src/lib/SafeBrowsing.py

```diff
@@ -0,0 +1,119 @@
+
+        update_existing_data = os.environ.get("CONNECTOR_UPDATE_EXISTING_DATA", "false")
+        if update_existing_data.lower() in ["true", "false"]:
+            self.update_existing_data = update_existing_data.lower()
+        else:
+            msg = f"Error when grabbing CONNECTOR_UPDATE_EXISTING_DATA environment variable: '{self.interval}'. It SHOULD be either `true` or `false`. `false` is assumed. "
+            self.helper.log_warning(msg)
+            self.update_existing_data = "false"
+
+    def google_safe_browsing(self, observable):
+        """Checks a domain against the Google Safe Browsing API."""
+
+        API_KEY = os.environ.get("GOOGLE_SAFE_BROWSING_API_KEY", "")
+        domain = observable["value"]
+        url = f"https://safebrowsing.googleapis.com/v4/threatMatches:find?key={API_KEY}"
+        payload = {
+            "client": {
+                "clientId": "OpenCTI_Connector",
+                "clientVersion": "1.0",
+            },
+            "threatInfo": {
+                "threatTypes": [
+                    "MALWARE",
+                    "SOCIAL_ENGINEERING",
+                    "UNWANTED_SOFTWARE",
+                    "POTENTIALLY_HARMFUL_APPLICATION",
+                    "THREAT_TYPE_UNSPECIFIED",
+                ],
+                "platformTypes": ["ANY_PLATFORM"],
+                "threatEntryTypes": [
+                    "URL",
+                    "EXECUTABLE",
+                    "THREAT_ENTRY_TYPE_UNSPECIFIED",
+                ],
+                "threatEntries": [{"url": domain}],
+            },
+        }
+        response = requests.post(
+            url,
+            json=payload,
+        )
+        if response.status_code == 200:
+            if response.json():
+                # self.helper.log_info(f"Domain {domain} flagged by Safe Browsing")
+
+                # Get the existing description and add the new one
+
+                if observable.get("x_opencti_description") is None:
+                    existing_description = ""
+                else:
+                    existing_description = (
+                        observable.get("x_opencti_description") + "\n"
+                    )
+
+                safe_browsing_output = response.json()
+
+                first_output = safe_browsing_output.get("matches")[0]
+
+                observable_description = f"Domain flagged by Safe Browsing \n + Threat Type: {first_output.get('threatType')} \n + Platform Type: {first_output.get('platformType')} \n + Threat Entry Type: {first_output.get('threatEntryType')}"
+
+                self.helper.api.stix_cyber_observable.update_field(
+                    id=observable["id"],
+                    input={
+                        "key": "x_opencti_description",
+                        "value": existing_description + observable_description,
+                    },
+                )
+
+                self.helper.api.stix_cyber_observable.add_label(
```

Comment:
As now, we need to send a bundle for playbook automation, here we need to create a STIX object using stix library

Reply:
We are not creating the Observable but only a label and updating description; Should I still need to create the observable again?

Reply:
When you do this, you are not creating a new observable, the mechanism is that if the observable already exists, it will do an update to complete the data to be enriched.

You can have more details on the OpenCTI documentation below:

- https://docs.opencti.io/latest/usage/deduplication/?h=upsert

Reply:
@helene-nguyen ; STIX2 doesn't have a Hostname object; How to tackle that

Reply:
@helene-nguyen ; changes are completed. Now sending them as stix bundles